Source/WebCore/rendering/RenderTheme.h

```diff
@@ -69,10 +69,20 @@ class RenderTheme {
     // "border" are set, or if the appearance is not supported by the theme.
     void adjustStyle(RenderStyle&, const Element*, const RenderStyle* userAgentAppearanceStyle);
 
-    // This method is called to paint the widget as a background of the RenderObject.  A widget's foreground, e.g., the
-    // text of a button, is always rendered by the engine itself.  The boolean return value indicates
+    virtual bool canCreateControlPartForRenderer(const RenderObject&) const { return false; }
+    RefPtr<ControlPart> createControlPartForRenderer(const RenderObject&) const;
+    ControlPart* controlPartForRenderer(const RenderBox&) const;
```

Comment:
is this `existingControlPartForRenderer` or `ensureControlPartForRenderer`?

Reply:
It is `ensureControlPartForRenderer` but it will return `ControlPart*` because only `Meter` is capable of creating its `MeterPart` on macOS right now.